app/models/treasury.rb

```diff
@@ -0,0 +1,522 @@
+        end
+
+        season_from_cards(snumber, season["description"], card_data, 'seasons')
+      end
+
+    elsif type == CLIPS
+      filepath = File.join(Rails.root, "app", "views", "treasuries", "data", "#{title}.yml")
+      @data = YAML.load(File.read(filepath))
+      @data["type"] = 'clips'
+      @data["treasury_url"] = "/collections/alistair-cooke"
+      @data["treasury_nice_title"] = "Alistair Cooke Masterpiece Collection"
+      @data["description"] = "While the Archives digitized and preserved Masterpiece Theatre, due to copyright we are unable to publicly post on-line full videos of the programs. However, as the introductions and closings of the episodes were filmed at GBH, we hope you enjoy these clips of Alistair Cooke as he brought viewers into the episodes and closed them out."
+      @data["posterImage"] = nil
+
+      season_data = Treasury.xml_docs.select { |x| Treasury.clip_assettype_from_xml(x) }.map { |xml| PBCore.new(xml) }.group_by(&:season_number)
+      @data["seasons"] = @data["seasons"].map do |season|
+        snumber = season["seasonNumber"].to_s
+
+        card_data = []
+        # one season's card data
+        if season_data[snumber]
+          card_data = season_data[snumber].map { |pb| card_from_pbcore(pb) }
+        end
+
+        season_from_cards(snumber, season["description"], card_data, 'clips')
+      end
+    else
+
+      # normalized, from url
+      miniseries_title = title
+      # get every pbcore record that shares this miniseries_title
+      minipbs = Treasury.xml_docs.select { |xml| miniseries_title == Treasury.normalize_mini_title(Treasury.miniseries_title_from_xml(xml)) }.map { |xml| PBCore.new(xml) }
+
+      # program number AKA episode number
+      miniseries_data = minipbs.group_by(&:program_number)
+
+      @data = {}
+      @data["type"] = 'episodes'
+
+      nice_miniseries_title = minipbs.first ? minipbs.first.miniseries_title : 'Miniseries Title Not Found'
+      @data["title"] = nice_miniseries_title
+
+      tseries = Treasury.treasury_series
+      home_treasury = tseries.keys.find { |treasury_title| tseries[treasury_title][:miniseries_titles].include?(@data["title"]) }
+
+      if home_treasury
+        @data["treasury_url"] = "/treasuries/#{home_treasury}"
+        @data["treasury_nice_title"] = tseries[home_treasury][:treasury_nice_title]
+      end
+
+      # storedty redundantly but thats how the cookie catalogs
+      miniseries_description  = minipbs.first ? minipbs.first.miniseries_description : ""
+      @data["description"] = miniseries_description
+      @data["seasons"] = []
+
+      miniseries_data.each do |episode_number, episode_pbs|
+        card_data = episode_pbs.map { |pb| card_from_pbcore(pb) }
+        # for miniseries page, a 'season' is ONE EPISODE
+        @data["seasons"] << season_from_cards(episode_number, nil, card_data, 'episodes')
+      end
+    end
+  end
+
+  def images
```

Comment:
Would it make sense to pull these and the treasury_series in to a yml file or something?